extensions/sql-database-projects/src/test/project.test.ts

```diff
@@ -1253,6 +1253,96 @@ describe('Project: sdk style project content operations', function (): void {
 	});
+
+	it('Should handle deleting empty folders', async function (): Promise<void> {
+		const testFolderPath = await testUtils.generateTestFolderPath();
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.newSdkStyleProjectSdkNodeBaseline, testFolderPath);
+
+		const project: Project = await Project.openProject(projFilePath);
+
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(0);
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(0);
+
+		// add an empty folder
+		await project.addFolderItem('folder1');
+
+		// verify folder was added
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(1);
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(0);
+		should(project.files.find(f => f.relativePath === 'folder1\\')).not.equal(undefined, 'folder1 should have been added');
+
+		// verify entry was added for the new empty folder in the sqlproj
+		let projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText.includes('<Folder Include="folder1\\" />')).equal(true, projFileText);
+
+		// delete the empty folder
+		await project.deleteFileFolder(project.files.find(f => f.relativePath === 'folder1\\')!);
+
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(0, 'folder1 should have been deleted');
+
+		// verify the folder entry was removed from the sqlproj and a Build Remove was not added
+		projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText.includes('<Folder Include="folder1\\" />')).equal(false, projFileText);
+		should(projFileText.includes('<Build Remove="folder1\\**" />')).equal(false, projFileText);
+	});
+
+	it('Should handle deleting not empty glob included folders', async function (): Promise<void> {
+		const testFolderPath = await testUtils.generateTestFolderPath();
+		projFilePath = await testUtils.createTestSqlProjFile(baselines.openSdkStyleSqlProjectBaseline, testFolderPath);
+		await testUtils.createDummyFileStructureWithPrePostDeployScripts(false, undefined, path.dirname(projFilePath));
+
+		const project: Project = await Project.openProject(projFilePath);
+
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(13);
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(3);
+
+		// delete a folder with contents
+		await project.deleteFileFolder(project.files.find(f => f.relativePath === 'folder2\\')!);
+
+		should(project.files.filter(f => f.type === EntryType.File).length).equal(8);
+		should(project.files.filter(f => f.type === EntryType.Folder).length).equal(2);
+
+		// verify the folder entry was removed from the sqlproj and a Build Remove was not added
+		const projFileText = (await fs.readFile(projFilePath)).toString();
+		should(projFileText.includes('<Build Remove="folder2\\**" />')).equal(false, projFileText);
```

Comment:
Same comment as before about confirming the absence of a very specific substring.